js-service/extras/legend_barcharts.js

```diff
@@ -0,0 +1,152 @@
+
+(function () {
+    'use strict';
+
+    angular
+        .module('flairbiApp')
+        .factory('legend', legend);
+
+    legend.$inject = ['common'];
+
+    function legend() {
```

Comment:
common has to be inject 
function legend(common) { 
please check the file for reference 
scheduler-dialog.controller.js